script.js

```diff
@@ -1,12 +1,216 @@
+  showSelector.appendChild(ShowAlloption);
+  allShows.forEach((show) => {
+    let option = document.createElement("option");
+    option.innerText = show.name;
+    option.value = show.id;
+    showSelector.appendChild(option);
+  });
+}
+
+//Search shows
+function searchShows() {
+  let searchTerm = event.target.value.toLowerCase();
+  filteredShows = allShows.filter((show) => {
+    return (show.name + show.summary).toLowerCase().includes(searchTerm);
+  });
+
+  displayCount.innerText = `Displaying ${filteredShows.length}/ ${allShows.length}`;
+
+  getShows(filteredShows);
+}
+
+//-----Getting Episodes Page
+
+/* 
+  In this section you're trying to get all episodes page
+  getEpisodes is called when a show is selected
+  makePageForEpisodes is called in different places to create the html page for episodes*/
+
+searchBar.addEventListener("keyup", () => searchEpisodes(allEpisodes));
+selectElement.addEventListener("change", handleSelect);
+//whenever a show is selected, we are converting to episodes page and hide any show parts
+function hideShowRelatedParts() {
+  selectElement.style.display = "";
+  searchBar.style.display = "";
+  showSearchBar.style.display = "none";
+}
+
+function getEpisodes() {
+  showId = event.target.value;
+  if (showId === "all") {
+    setup();
+  } else {
+    //hiding other show parts
+    hideShowRelatedParts();
+
+    fetch(`https://api.tvmaze.com/shows/${showId}/episodes`)
+      .then((response) => {
+        return response.json();
+      })
+      .then((data) => {
+        makePageForEpisodes(data);
+        buildEpisodeOptions(data);
+      })
+      .catch((error) => {
+        console.log(error);
+      });
+  }
+}
+
+function makePageForEpisodes(episodesList) {
+  containerEl.innerHTML = "";
+  episodesList.forEach((episode) => createCardForEpisodes(episode));
+}
+
+function createCardForEpisodes(episode) {
+  const cardEl = document.createElement("div");
+  cardEl.classList = "card";
+  const titleEl = document.createElement("h3");
+  titleEl.classList = "episode-info";
+  const summaryEl = document.createElement("p");
+  const img = document.createElement("img");
+  titleEl.textContent =
+    episode.name + getSeasonAndEpisode(episode.season, episode.number);
+
+  img.setAttribute("src", episode.image ? episode.image.medium : unfoundImgSrc);
+  summaryEl.innerHTML = episode.summary;
+  cardEl.append(titleEl, img, summaryEl);
+  containerEl.appendChild(cardEl);
+}
+
+//formating eposide name
+function getSeasonAndEpisode(season, episode) {
+  if (season.toString().length < 2) {
+    season = "0" + season;
+  }
+
+  if (episode.toString().length < 2) {
+    episode = "0" + episode;
+  }
+  return ` -S${season}E${episode}`;
+
+  //   `${episode.name}` +
+  //     ` -S${episode.season
+  //       .toString()
+  //       .padStart(2, "0")}E${episode.number.toString().padStart(2, "0")}`;
+}
+
+function searchEpisodes(allEpisodes) {
+  let searchTerm = event.target.value.toLowerCase();
```

Comment:
The issue here is that you are passing `allEpisodes` to the function, but then you are expecting an `event`. Remember, you should only refer to an event if you defined is a function variable. If you want to have access to both the episodes and the change event, then you can make this function take two arguments.

I will send a separate PR to show you the changes you need to make.